biopandas/mmcif/pandas_mmcif.py

```diff
@@ -88,6 +88,23 @@ def fetch_mmcif(self, pdb_code: str):
         self.mmcif_path, self.mmcif_text = self._fetch_mmcif(pdb_code)
         self._df = self._construct_df(text=self.mmcif_text)
         return self
+    
+    def fetch_af2(self, uniprot_id: str):
```

Comment:
Cool idea! I wonder if we should maybe extend the user-facing `fetch_mmcif` method instead though? Just so in case there are future methods, like alphafold3 and others, we don't have to write a new method each time. 

We can keep the private `_fetch_af2` of course, but for the user-facing `fetch_` method, maybe we can keep `fetch_mmcif` and just change the source? Something like

    fetch_af2(uniprot_id=..., source="pdb") # default
    fetch_af2(uniprot_id=..., source="alphafold2") 

What do you think?

Reply:
That makes sense too. Happy to make the change if you'd prefer that.

Reply:
On reflection, I think if we proceed with adding the source argument to the the `fetch_mmcif` (or `fetch_pdb`) methods, it would make most sense to rename the `pdb_code` argument to something like `id` or `accession` which would make it a breaking change.

The alternative is to have three args: `pdb_code`, `uniprot_id`, and `source` and then some logic in the function to validate the input (e.g. using a PDB code to get a structure from AF2 wouldn't work).

Reply:
Hm, good point. It's bad to introduce breaking changes, but I assume most people hopefully used it as `fetch_pdb("...")` rather than `fetch_pdb(pdb_code"...")`. In the long run, it might be better to name it `access_code`, `id`, or `id_code` I'd say.

Or, we could maybe keep `pdb_code` with a deprecation warning for now and have it behave like the `access_code`?

Reply:
I think in this case I'd lean on [PEP20](https://peps.python.org/pep-0020/)'s "explicit is better than implicit" and vote for the three arg approach as it doesn't break any existing code and probably enhances readability and traceability.

Reply:
You mean

    fetch_pdb(pdb_code="...", uniprot_id="...", source="...")

Ok, fair regarding avoiding breaking existing code. We just have to add a simple check that one of the two, `pdb_code` and `uniprot_id` needs to be `None` so there is no ambiguity or conflict. Maybe silly question, but can you fetch a PDB file from PDB via `uniprot_id`?

Reply:
Yep, I'll make the change now :)

You certainly can but requires some trickery as it's not cleanly bijective e.g. the same protein crystallised under different conditions, or a structure containing multiple interactors, and ofc not all uniprots have structures. I've used [ssbio](https://ssbio.readthedocs.io/en/latest/) for this in the past - essentially they BLAST the sequence retrieved from UniProt against chains in the PDB and provide a ranking based on quality/resolution, identity etc

Reply:
Thanks! So, how would be handle things like

    fetch_pdb(uniprot_id="something", source="rcsb/pdb") 

Should we have checks that if people use a rcsb/pdb as a `source` they should be using a `pdb_code` instead?

Reply:
I think this does the trick:

```python
    def fetch_mmcif(self, pdb_code: Optional[str] = None, uniprot_id: Optional[str] = None, source: str = "pdb"):
        """Fetches mmCIF file contents from the Protein Databank at rcsb.org.

        Parameters
        ----------
        pdb_code : str, optional
            A 4-letter PDB code, e.g., `"3eiy"` to retrieve structures from the PDB. Defaults to `None`.

        uniprot_id : str, optional
            A UniProt Identifier, e.g., `"Q5VSL9"` to retrieve structures from the AF2 database. Defaults to `None`.

        source : str
            The source to retrieve the structure from (`"pdb"` or `"alphafold2"`). Defaults to `"pdb"`.

        Returns
        ---------
        self

        """
        # Sanitize input
        invalid_input_identifier_1 = pdb_code is None and uniprot_id is None
        invalid_input_identifier_2 = pdb_code is not None and uniprot_id is not None
        invalid_input_combination_1 = uniprot_id is not None and source == "pdb"
        invalid_input_combination_2 = pdb_code is not None and source == "alphafold2"

        if invalid_input_identifier_1 or invalid_input_identifier_2:
            raise ValueError("Please provide either a PDB code or a UniProt ID.")

        if invalid_input_combination_1 or invalid_input_combination_2:
            raise ValueError(f"Please use a valid accession ID for your chosen structural database source ({source}).")

        if source == "pdb":
            self.mmcif_path, self.mmcif_text = self._fetch_mmcif(pdb_code)
        elif source == "alphafold2":
            self.mmcif_path, self.mmcif_text = self._fetch_af2(uniprot_id)
        else:
            raise ValueError(f"Invalid source: {source}. Please use one of 'pdb' or 'alphafold2'.")

        self._df = self._construct_df(text=self.mmcif_text)
        return self



```

Reply:
That'd do the trick! I wonder if we should make 

```python
        if invalid_input_combination_1 or invalid_input_combination_2:
            raise ValueError(f"Please use a valid accession ID for your chosen structural database source ({source}).")
```
more clear though? E.g.,

```python
        if invalid_input_combination_1 :
            raise ValueError("Please use a `pdb_code` instead of `unitprod_id` for source="pdb".)
        elif invalid_input_combination_2 :
            raise ValueError("Please use a `uniprot_id` instead of `pdb_code` for source="alphafold2".)
```

Reply:
Yep, good idea!